src/test/java/cubecart/test/BaseTest.java

```diff
@@ -2,8 +2,56 @@
 
 import cubecart.utils.SessionUtil;
 import org.apache.log4j.Logger;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
 
 public class BaseTest {
-    SessionUtil sessionUtil = SessionUtil.getInstance();
-    static Logger log = Logger.getLogger(BaseTest.class);
+
+    ChromeDriver driver;
+
+
+    public void setUp(){
+        System.out.println("Setting up resources for the test...");
+    }
+
+
+    public  void openBrowser() {
+        driver = new ChromeDriver();
+        driver.manage().window().maximize();
+        driver.get("https://cubecartuat.unitedcoderapp.com/admin_d9nchr.php");
+    }
+
+
+
+    public  void login(String userName, String password) {
+        WebElement userNameField = driver.findElement(By.id("username"));
+        userNameField.sendKeys(userName);
+        WebElement passwordField = driver.findElement(By.id("password"));
+        passwordField.sendKeys(password);
+        WebElement loginButton = driver.findElement(By.id("login"));
+        loginButton.click();
+    }
+
+    public  void logOut() {
+        WebElement logOutLink = driver.findElement(By.linkText("Log Out"));
+        logOutLink.click();
+    }
+
+    public  void closeBrowser() {
+        System.out.println("Tearing down resources after the test...");
+        driver.close();
+        driver.quit();
+    }
+
```

Comment:
here you can combine setup & openbrowser methods into one method and call it setup.  and use testng beforeMethod annotation. 
you need to have another method called tearDown() and use afterMethod annotation.  please work with @Babur101 for the driver class implementation
your baseTest should be something similar to below screenshot.  
<img width="831" alt="Screenshot 2024-02-26 at 16 06 40" src="https://github.com/A-Diliyaer/Cube24/assets/65502064/75df6fe2-d8b2-4f89-9856-84ca70efc0c7">